src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1221,30 +1230,33 @@ impl<T: 'static> EventProcessor<T> {
     }
 
     fn handle_pressed_keys<F>(
-        &self,
+        wt: &super::EventLoopWindowTarget<T>,
         window_id: crate::window::WindowId,
         state: ElementState,
+        mod_keymap: &ModifierKeymap,
+        device_mod_state: &mut ModifierKeyState,
         callback: &mut F,
     ) where
         F: FnMut(Event<'_, T>),
     {
-        let wt = get_xtarget(&self.target);
-
         let device_id = mkdid(util::VIRTUAL_CORE_KEYBOARD);
-        let modifiers = self.device_mod_state.modifiers();
-
-        // Get the set of keys currently pressed and apply Key events to each
-        let keys = wt.xconn.query_keymap();
+        let modifiers = device_mod_state.modifiers();
 
-        for keycode in &keys {
-            if keycode < 8 {
-                continue;
-            }
-
-            let scancode = (keycode - 8) as u32;
+        // Get the set of keys currently pressed and apply Key events to each and also updating
+        // the pressed modifiers.
```

Comment:
```suggestion
        // Update modifiers state and emit key events based on which keys are currently pressed.
```

Shorter and a little easier to read I think. The `and also updating` part sounds a bit strange.

Reply:
Yeah, I agree. Since I was the one who wrote that part, I guess it's fine if I commit it.